File: src/SlidingWindow/Exercise1004.java
package SlidingWindow;
public class Exercise1004 {
    class Solution {
        public int longestOnes(int[] nums, int k) {
            int left = 0, right = 0;
            int surplus = k;
            int max = 0;
            while (right < nums.length) {
                if (nums[right] == 0) {
                    surplus--;
                }
                if (surplus < 0) {
                    max = Math.max(max, right - left);

                    if (nums[left] == 0) {
                        surplus++;

                    }
                    left++;
                }

                right++;
            }
            max = Math.max(max, right - left);
            return max;
        }

    }

    public static void main(String[] args) {
        Exercise1004 exercise = new Exercise1004();
        Solution solution = exercise.new Solution();
        int[] nums = new int[] { 0, 0, 1, 1, 0, 0, 1, 1, 1, 0, 1, 1, 0, 0, 0, 1, 1, 1, 1 };
        System.out.println(solution.longestOnes(nums, 3));
    }

}

File: src/SlidingWindow/Exercise1004.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution
{
public:
    int longestOnes(vector<int> &nums, int k)
    {
        int n = nums.size();
        int maxCount = 0;
        int left = 0, right = 0;
        int count=0;
        while(right<n){
            if(nums[right]==0){
                count++;
            }
            while(count>k){
                if(nums[left]==0){
                    count--;
                }
                left++;
            }
            maxCount = max(maxCount, right - left + 1);
            right++;
        }

        return maxCount;
    }
};

int main()
{
    vector<int> nums{1, 1, 1, 0, 0, 0, 1, 1, 1, 1, 0};
    Solution sl;
    cout << sl.longestOnes(nums, 2);
}